bftengine/include/bftengine/IStateTransfer.hpp

```diff
@@ -68,6 +69,12 @@ class IStateTransfer {
   // Return the internal state (member variables, etc...) of the state transfer module as a string. This is used by the
   // diagnostics subsystem.
   virtual std::string getStatus() { return ""; };
+
+  // Registers a function that is called every time State Transfer completes.
+  // Accepts the checkpoint number as a parameter.
+  // Callbacks must not throw.
+  // Multiple callbacks can be added.
+  virtual void addOnTransferringCompleteCallback(std::function<void(int64_t)>) = 0;
```

Comment:
Why an int64_t and not uint64_t?

Reply:
I think the existing callback uses int64_t, but I might be mistaken. Will double check when around a PC and change accordingly. Thanks for noticing.

Reply:
Changed to uint64_t.